src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -1012,22 +1130,46 @@ SubCmdValidate::execute(const SubCmdOptions& _options) const
 
   XBU::collect_devices(deviceNames, true /*inUserDomain*/, deviceCollection);
 
-  //validate all devices
-  boost::property_tree::ptree _ptValidate;
-  boost::property_tree::ptree _ptDevCollectionTestSuite;
-  if(!json)
-    std::cout << boost::format("Starting validation for %d devices\n\n") % deviceCollection.size();
+  // Collect all of the tests of interests
+  std::vector<TestCollection *> testObjectsToRun;
 
-  for(auto const& dev : deviceCollection) {
-    run_test_suite_device(dev, _ptDevCollectionTestSuite, json, quick);
+  for (unsigned index = 0; index < testSuite.size(); ++index) {
+    if (testsToRun[0] == "all") {
+      testObjectsToRun.push_back(&testSuite[index]);
+      continue;
+    }
+
+    if (testsToRun[0] == "quick") {
```

Comment:
else if?

Reply:
Don't believe one is necessary here.   Let me explain further.
This condition is part of a loop of tests that the user has requested to be ran.  The key here is that ALL the possible tests are examined to see if they match one of the tests in the user collection (not visa-versa).  This allows for a more efficient search with the following conditions, especially since that the user set of tests have already been qualified, no need to do it again.   

Besides, the "continue" and "break" remove the need for the "else if" :)